src/partialBindTo.test.ts

```diff
@@ -0,0 +1,71 @@
+import { partialBindTo } from './partialBindTo';
+
+const fn = (x: number, y: number, z: number) => `${x}, ${y}, and ${z}`;
+
+describe('runtime', () => {
+  describe('data-first', () => {
+    test('should partially apply 0 args', () => {
+      expect(partialBindTo([], fn)(1, 2, 3)).toBe(fn(1, 2, 3));
+    });
+
+    test('should partially apply 1 arg', () => {
+      expect(partialBindTo([1], fn)(2, 3)).toBe(fn(1, 2, 3));
+    });
+
+    test('should partially apply all args', () => {
+      expect(partialBindTo([1, 2, 3], fn)()).toBe(fn(1, 2, 3));
+    });
+  });
+
+  describe('data-last', () => {
+    test('should partially apply 0 args', () => {
+      expect(partialBindTo(fn)([])(1, 2, 3)).toBe(fn(1, 2, 3));
+    });
+
+    test('should partially apply 1 arg', () => {
+      expect(partialBindTo(fn)([1])(2, 3)).toBe(fn(1, 2, 3));
+    });
+
+    test('should partially apply all args', () => {
+      expect(partialBindTo(fn)([1, 2, 3])()).toBe(fn(1, 2, 3));
+    });
+  });
+});
+
+describe('typing', () => {
+  describe('data-first', () => {
+    test('should correctly type 0 partial args', () => {
+      expectTypeOf(partialBindTo([], fn)).toEqualTypeOf<
+        (x: number, y: number, z: number) => string
+      >();
+    });
+
+    test('should correctly type 1 partial arg', () => {
+      expectTypeOf(partialBindTo([1], fn)).toEqualTypeOf<
+        (y: number, z: number) => string
+      >();
+    });
+
+    test('should correctly type all partial args', () => {
+      expectTypeOf(partialBindTo([1, 2, 3], fn)).toEqualTypeOf<() => string>();
+    });
+  });
+
+  describe('data-last', () => {
```

Comment:
dataLast should also test inference via `pipe`, as those work a bit differently to simple dataLast invocations. Because dataLast is mainly used in pipes those are the most important cases to test.